docs/getting-started/actions.md

```diff
@@ -2,73 +2,144 @@
 title: Actions
 ---
 
-You can update Variables through Dara actions which can be passed to any component prop accepting an `Action`.
+Actions are a way to add interactivity to your Dara app in response to some user interaction. Components accept actions as arguments to their callbacks annotated with type `Action`, usually named `onclick`, `onchange` or similar. Actions can be used to update variables, trigger notifications, navigate to a different page, and more.
 
-## `UpdateVariable`
+A valid argument to an action parameter is one of the following three options:
 
-The `dara.core.interactivity.actions.UpdateVariable` action can trigger the update of a `Variable`, `DataVariable` or `UrlVariable`.
+- an instance of an `@action`-decorated function (as in the example above)
+- an instance of an individual static action object
+- a list of a combination of the above two options
 
-It takes the following arguments:
+In this section you will learn all the different ways to create actions and how to use them.
 
-- `resolver`: a function to resolve the new value for the `Variable`, `DataVariable` or `UrlVariable`. The resolver takes one argument: a context of type `UpdateVariable.Ctx`. The new value is given by the component and can be obtained with `ctx.inputs.new`. While the current value, or now previous value of the variable you are updating, can be obtained with `ctx.inputs.old`.
-- `variable`: the `Variable`, `DataVariable` or `UrlVariable` to update with a new value upon triggering the action
-- `extras`: any extra variables to resolve and pass to the resolver function, you can obtain a list of the resolved values of all the extras passed with `ctx.extras` in the resolver function
+## `@action` decorator
+
+The primary way to create actions in Dara is the `@action` decorator. This decorator takes a function and returns an action that can be passed to a component's callback.
+It injects an `ActionCtx` object as the first argument of the function, which contains the input sent from the component and exposes action methods to update the state of the application.
+
+:::note
+
+The action methods on `ActionCtx` are `async` and must be `await`-ed, so your decorated function should be `async` as well.
```

Comment:
Tbh on the surface level there isn't much you need to know to use this API - just mark your action as `async` and `await` methods which are marked `async`. I appreciate it's hard for me to put myself in a data scientist's shoes so not sure how to explain it to them well. 
Do you have a particular resource that you think would work well? I can think of either a generic introduction like this blogpost https://realpython.com/async-io-python/ or FastAPI's section https://fastapi.tiangolo.com/async/#in-a-hurry (we've linked to this in other places).

Reply:
agreed that they don't need to know much and I think that's a good thing! but I can see it peaking their curiosity as to how things like that work and I don't want to give the impression that we just expect them to be familiar with it. I like the first link as it puts things into context of other concurrency and paralellism techniques which a data scientist probably will be familiar with.

_however_ if you think it might be distracting and lead the reader astray during an important part of the docs then I could be persuaded to not link anything

Reply:
Fair enough, thanks. This is just a 'note' section so shouldn't be too distracting, will link the blogpost then.